src/main/java/sg/ncl/MainController.java

```diff
@@ -2560,12 +2561,130 @@ public String rejectTeam(
         return "redirect:/admin";
     }
 
+    @RequestMapping("/admin/teams/{teamId}")
+    public String setupTeamRestriction(
+            @PathVariable final String teamId,
+            @RequestParam(value = "action", required=true) final String action,
+            final RedirectAttributes redirectAttributes,
+            HttpSession session) throws IOException
+    {
+        final String logMessage = "Updating restriction settings for team {}: {}";
+
+        // check if admin
+        if (!validateIfAdmin(session)) {
+            log.warn(logMessage, teamId, PERMISSION_DENIED);
+            return NO_PERMISSION_PAGE;
+        }
+
+        Team2 team = invokeAndExtractTeamInfo(teamId);
+
+        // check if team is approved before restricted
+        if ("restrict".equals(action) && team.getStatus().equals(TeamStatus.APPROVED.name())) {
+            return restrictTeam(team, redirectAttributes);
+        }
+        // check if team is restricted before freeing it back to approved
+        else if ("free".equals(action) && team.getStatus().equals(TeamStatus.RESTRICTED.name())) {
+            return freeTeam(team, redirectAttributes);
+        } else {
+            log.warn(logMessage, teamId, "Cannot " + action + " team with status " + team.getStatus());
+            redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + "Cannot " + action + " team " + team.getName() + " with status " + team.getStatus());
+            return "redirect:/admin";
+        }
+    }
+
+    private String restrictTeam(final Team2 team, RedirectAttributes redirectAttributes) throws IOException {
+        log.info("Restricting team {}", team.getId());
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(
+                properties.getSioTeamsStatusUrl(team.getId(), TeamStatus.RESTRICTED),
+                HttpMethod.PUT, request, String.class);
+        String responseBody = response.getBody().toString();
+
+        if (RestUtil.isError(response.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            String logMessage = "Failed to restrict team {}: {}";
+            switch (exceptionState) {
+                case TEAM_NOT_FOUND_EXCEPTION:
+                    log.warn(logMessage, team.getId(), "Team not found");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Team not found" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)